traffic_ops/app/bin/checks/ToDSCPCheck.pl

```diff
@@ -105,179 +111,231 @@
-               }
-            } else {
-               $successful = 0;
-               TRACE "Fail deliveryService: ".$ds->{profileName};
-               if ($sslg) {
-                  my @tmp = ($fqdn,$check_name,$chck_lng_nm,'FAIL',$status,
-                             $target,$header,"Expected DSCP value of $ds->{dscp} got $dscp_found");
-                  syslog(LOG_ERR, "hostname=%s check=%s name=\"%s\" result=%s status=%s target=%s url=%s msg=\"%s\"", @tmp);
                }
-            }
-
-
-            ## check ipv6
-            # TODO "http://" should be a var so that we can also check https
-            # TODO "80" should be var when we add https
-            $url = "http://".$ip.":".$tcpPort.$ds->{checkPath};
-
-            TRACE "About to check header: ".$header." url: ".$url;
-
-            if ($force == 0) {
-               $dscp_found = &get_dscp( $url, $ip6, $cms_int, $header, "ipv6");
-            } elsif ($force == 1) {
-               $dscp_found = -1;
-            } elsif ($force == 2) {
-               $dscp_found = $ds->{dscp};
-            } elsif ($force == 3) {
-               $dscp_found = $ds->{dscp} + 1;
-            }
-            $target = $ds->{xmlId}.":ipv6";
-            if ($dscp_found == -1) {
-               $successful = 0;
-               TRACE "Failed deliveryService: ".$ds->{profileName};
-               if ($sslg) {
-                  my @tmp = ($fqdn, $check_name, $chck_lng_nm, 'FAIL',$status,
-                             $target,$header,"Unable to connect to edge server");
-                  syslog(LOG_INFO, "hostname=%s check=%s name=\"%s\" result=%s status=%s target=%s url=%s msg=\"%s\"", @tmp);
-               }
-            } elsif ( $dscp_found == $ds->{dscp} ) {
-               TRACE "Success deliveryService: ".$ds->{profileName};
-               if ($sslg) {
-                  my @tmp = ($fqdn,$check_name,$chck_lng_nm,'OK',$status,
-                             $target,$header);
-                  syslog(LOG_INFO, "hostname=%s check=%s name=\"%s\" result=%s status=%s target=%s url=%s msg=\"\"", @tmp);
-               }
-            } else {
-               $successful = 0;
-               TRACE "Fail deliveryService: ".$ds->{profileName};
-               if ($sslg) {
-                  my $target = $ds->{xmlId}.":ipv6";
-                  my @tmp = ($fqdn, $check_name, $chck_lng_nm, 'FAIL',$status,
-                             $target,$header,"Expected DSCP value of $ds->{dscp} got $dscp_found");
-                  syslog(LOG_ERR, "hostname=%s check=%s name=\"%s\" result=%s status=%s target=%s url=%s msg=\"%s\"", @tmp);
+               foreach my $ip (@ip_addrs) {
+                  TRACE "TRACE checking " . $ip;
+                  if ( !defined($ip) ) {
+                     TRACE "TRACE IP not defined. Should go to next ip";
+                     next;
+                  } elsif ( $ip =~ m/\./ ) {
+                     DEBUG "DEBUG ipv4: " . $ip_addrs[0];
```

Comment:
shouldn't this be `DEBUG "DEBUG ipv4: " . $ip`? Seems like this will always log the first IP address regardless of where you actually are in the iteration. Also, IPv6 addresses are not being logged in this way - should they be?

Reply:
Fixed. IPv6 is a few lines down.